src/test/resources/com/puppycrawl/tools/checkstyle/checks/trailingcomment/InputTrailingCommentPastFalsePositive.java

```diff
@@ -0,0 +1,136 @@
+
+/**
+ * Config:
+ * default
+ */
+public class InputTrailingCommentPastFalsePositive {
+    public static boolean mybool = true; // violation
+    /* FROM NOW ON EVERYTHING IN THIS INPUT FILE SHOULD TRIGGER NO TRAILING COMMENT VIOLATION!!! */
+    public static void main(String[] args) {
+        if (mybool) { /* Multiline comment with
+                        embedded'       'tab
+                 (not end line -> ok) */ char myChar = ' ';
+        }
+    }
+
+
+    /** should.be.ok **/
+    private boolean b = false;
+
+    /** should 'also' be 'ok' **/
+    private String s1;
+
+    /* comment here is ok */ void foo1() {
+        this.s1 = "abc";
+        if (s1 == null) {
+
+        }
+        /*** THIS SHOULD BE OK ***/
+        else {
+            b = true;
+        }
+        /***/
+    }
+
+    /*this is also ok*/ void foo2() {
+        this.b = true;
+        /*ok*/ String s3 = "\u03bcs";
+        if (b){/*okok*/}
+        try {
+            throw new NullPointerException();
+        } catch (NullPointerException e)
+        /*okok*/ {
+        }
+    }
+
+    void foo3 /**ok*/ () {}
+
+    public/*15*/void/*16*/someVeryVeryLongFunctionNameWithLotsOfComments()/*49*/{/*
+    21
+    */}
+
+    // Some one line comments above
+    // Then one multiline comment below
+
+    /* Should be ok */
+    private String s2 = "abc";
+
+    /**
+     * javadoc comment
+     * @param a a number
+     * @param b another number
+     * @param c another number
+     * @return a number :)
+     */
+    /* Block comment right after javadoc */int foo4(int a, int b, int c) {
+        if (a == b) {
+            s2 = c + "1";
+        }
+        return 1;
+    }
+
+
+    /* This is just a test!!!:                                                 //indent:0 exp:0
+     * This test-input is intended to be checked using following configuration:   //indent:1 exp:1
+     *                                                                            //indent:1 exp:1
+     * arrayInitIndent = 4                                                        //indent:1 exp:1
+     * basicOffset = 4                                                            //indent:1 exp:1
+     * braceAdjustment = 0                                                        //indent:1 exp:1
+     * caseIndent = 4                                                             //indent:1 exp:1
+     * forceStrictCondition = false                                               //indent:1 exp:1
+     * lineWrappingIndentation = 4                                                //indent:1 exp:1
+     * tabWidth = 4                                                               //indent:1 exp:1
+     * throwsIndent = 4                                                           //indent:1 exp:1
+     *                                                                            //indent:1 exp:1
+     *                                                                             //indent:1 exp:1
```

Comment:
this config also looks confusing with indentation and so on.
Please change it in case it shows some test case or remove completely

Reply:
fixed